editor/src/components/editor/wrap-in-callbacks.ts

```diff
@@ -1,33 +1,227 @@
+  wrapperUid: string,
+  indexPosition: IndexPosition,
+): CanvasCommand[] {
+  const parentType = autoLayoutParentAbsoluteOrStatic(
+    metadata,
+    allElementProps,
+    elementPathTrees,
+    parentPath.intendedParentPath,
+    'prefer-absolute',
+  )
+
+  const parentFrame = nullIfInfinity(
+    MetadataUtils.getFrameInCanvasCoords(parentPath.intendedParentPath, metadata),
+  )
+  if (parentFrame == null) {
+    return []
+  }
+
+  const desiredFrame: CanvasRectangle = canvasRectangle({
+    width: childrenBoundingFrame.width,
+    height: childrenBoundingFrame.height,
+    x: childrenBoundingFrame.x - parentFrame.x,
+    y: childrenBoundingFrame.y - parentFrame.y,
+  })
+
+  const props = jsxAttributesFromMap({
+    style: jsExpressionValue(getWrapperStyle(parentType, desiredFrame), emptyComments),
+  })
+  return [
+    addElement('always', parentPath, jsxElement('div', wrapperUid, props, []), {
+      indexPosition: indexPosition,
+    }),
+  ]
+}
+
+function getCommandsForChild(
+  metadata: ElementInstanceMetadataMap,
+  elementPathTrees: ElementPathTrees,
+  childrenBoundingFrame: CanvasRectangle,
+  newParentPath: ElementPath,
+  path: ElementPath,
+): CanvasCommand[] {
+  const instance = MetadataUtils.findElementByElementPath(metadata, path)
+  if (instance == null) {
+    return []
+  }
+
+  const frame = nullIfInfinity(instance.globalFrame) ?? zeroCanvasRect
+
+  return [
+    ...sizeToVisualDimensions(metadata, elementPathTrees, path),
+    ...setElementTopLeft(instance, {
+      top: frame.y - childrenBoundingFrame.y,
+      left: frame.x - childrenBoundingFrame.x,
+    }),
+    setProperty('always', path, PP.create('style', 'position'), 'absolute'),
+    reparentElement('always', path, childInsertionPath(newParentPath)),
+  ]
+}
+
+function getWrapperStyle(
+  parentType: ReparentStrategy,
+  desiredFrame: CanvasRectangle,
+): CSSProperties {
+  const style: CSSProperties = {
+    contain: 'layout',
+    width: desiredFrame.width,
+    height: desiredFrame.height,
+  }
+  switch (parentType) {
+    case 'REPARENT_AS_STATIC':
+      return style
+    case 'REPARENT_AS_ABSOLUTE':
+      return {
+        ...style,
+        position: 'absolute',
+        top: desiredFrame.y,
+        left: desiredFrame.x,
+      }
+    default:
+      assertNever(parentType)
   }
 }
 
-export function useWrapInDiv(): (event: React.MouseEvent<Element>) => void {
-  return useWrapInElement(makeWrapperDiv)
+export function useWrapInDiv(): () => void {
+  const dispatch = useDispatch()
+  const editorRef = useRefEditorState((store) => store.editor)
+
+  const wrapInDiv = React.useCallback(() => {
+    const result = wrapInDivCommands(
+      editorRef.current.jsxMetadata,
+      editorRef.current.elementPathTree,
+      editorRef.current.allElementProps,
+      editorRef.current.projectContents,
+      editorRef.current.selectedViews,
+    )
+
+    if (isLeft(result)) {
+      return
+    }
+
+    dispatch([applyCommandsAction(result.value)])
```

Comment:
maybe nitpicking, but this code is _almost_ structured like an inspector strategy, it might as well be one, with a `executeFirstApplicableStrategy` call instead of this `applyCommandsAction`

Reply:
The reason I didn't use `executeFirstApplicableStrategy` was because `wrapInDivCommands` takes a `projectContents` param, and for that `executeFirstApplicableStrategy`'s signature would have had to be changed too, and I wanted to keep the PR scoped to wrapping. We can make this change if you think it's important though

Reply:
<img width="790" alt="image" src="https://github.com/concrete-utopia/utopia/assets/2226774/8177f9c4-66c9-4fba-9df0-c915a9ab02b1">

most new inspector strategies take the extra params in their factory function, you could add projectContents there!

--------------------------

BTW this two sources of params actually look pretty confusing inside the strategy code:
<img width="453" alt="image" src="https://github.com/concrete-utopia/utopia/assets/2226774/e87ed30b-3c98-4b6b-a253-f2329bb18fff">

For the Canvas strategies we already moved to a setup where the `apply()` function is almost parameterless. I was thinking that we could do the same for inspector strategies, the `strategy()` function could have zero parameters, and only pass in the params to the strategy's factory function. 

The end result would be something like this:
```
executeFirstApplicableStrategy(dispatch, [
  moveInspectorStrategy(
    projectContentsRef.current,
    metadataRef.current,
    selectedViewsRef.current,
    frameUpdate.edgeMovement,
  ),
  resizeInspectorStrategy(
    projectContentsRef.current,
    metadataRef.current,
    selectedViewsRef.current,
    frameUpdate.edgeMovement,
  ),
])
```

Reply:
implemented in https://github.com/concrete-utopia/utopia/pull/4427/commits/041d4294a4a206966c8b7fe2f481fcc7fe79206b